src/main/java/com/cloudbees/jenkins/support/configfiles/TemporalFileContent.java

```diff
@@ -0,0 +1,36 @@
+package com.cloudbees.jenkins.support.configfiles;
+
+import com.cloudbees.jenkins.support.api.FileContent;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Temporal file content, auto-deleted after {@link #writeTo(OutputStream)}.
+ */
+public class TemporalFileContent extends FileContent {
+
+    private File f;
+
+    public TemporalFileContent(String name, File file) {
+        super(name, file);
+        f = file;
+    }
+
+    @Override
+    public void writeTo(OutputStream os) throws IOException {
+        super.writeTo(os);
+        delete();
+    }
+
+    private void delete() {
+        if(!f.delete()) {
```

Comment:
We probably really want to insist that those sensitive temporary config files gets deleted, hence better use either `hudson.Util#deleteFile` ideally, or at least `hudson.Util#tryOnceDeleteFile` ?